src/app/location/location.component.ts

```diff
@@ -0,0 +1,79 @@
+import { Component, OnInit, OnDestroy } from "@angular/core";
+import { HttpClass } from "../http.service";
+import { ICountryInfo } from "../interface/interface";
+import { ActivatedRoute } from "@angular/router";
+import { Subscription } from "rxjs";
+
+@Component({
+  selector: "app-location",
+  templateUrl: "./location.component.html",
+  styleUrls: ["./location.component.scss"],
+})
+export class LocationComponent implements OnInit, OnDestroy {
+  countries: ICountryInfo[];
+  mainData: ICountryInfo[];
+  latitude: number;
+  longitude: number;
+  isLoading: boolean = true;
+  error = null;
+  labels = ["Confirmed", "Recovered", "Deaths"];
+  period = [0, 7, 15, 30];
+  name: string;
+  temp: number;
+  searchText: string;
+  subs: Subscription[] = [];
+
+  constructor(private http: HttpClass, private route: ActivatedRoute) {}
+
+  ngOnInit(): void {
+    this.subs.push(
+      this.http.getAllCountries().subscribe((data) => {
+        this.countries = data.sort((a: ICountryInfo, b: ICountryInfo) =>
+          a.Country > b.Country ? 1 : -1
+        );
+      })
+    );
+
+    let id = "sebia";
+    this.route.params.subscribe((params) => (id = params["id"]));
+    this.dataForCountry(id);
+  }
+
+  onClick(data: string) {
```

Comment:
onClick naming is not a good one, what does this method do?

Reply:
This method displays data for country that's been clicked (triggers this.dataForCountry(data))